apps/estuary/lib/estuary_web/live/streaming_event_live_view.ex

```diff
@@ -20,18 +40,41 @@ defmodule EstuaryWeb.StreamingEventLiveView do
      assign(socket,
        events: [],
        no_events: "Waiting For The Events!",
-       order: {"create_ts", "desc"},
+       search_text: nil,
        params: %{}
      )}
   end
 
   def handle_info(%{topic: @updated_event_stream, payload: %{events: events}}, socket) do
     updated_events =
       (List.wrap(events) ++ List.wrap(socket.assigns.events))
-      |> Enum.filter(&(!is_nil(&1)))
-      |> Enum.take(1000)
-      |> Enum.sort(&(&1["create_ts"] >= &2["create_ts"]))
+      |> refresh_events(socket.assigns.search_text)
 
     {:noreply, assign(socket, :events, updated_events)}
   end
+
+  def handle_params(params, _uri, socket) do
+    filtered_events = refresh_events(socket.assigns.events, params["search"])
```

Comment:
The way this is written, we are dropping events that come in through kafka that don't show up with the current search text. So if you delete whatever search text is there, there will be some events that will never show up. Im assuming we want to show all of the events that have come through regardless of what is in the search box since the previous reload?

Reply:
We don't care about that much, so I think it is fine if something is dropped